chat/memberFilters.ts

```diff
@@ -0,0 +1,208 @@
+import { Channel } from './interfaces';
+import { CharacterAnalysis, Matcher } from '../learn/matcher';
+import { Gender, Scoring } from '../learn/matcher-types';
+
+export const genderOptions: string[] = [
+  'Female',
+  'Male',
+  'Herm',
+  'Shemale',
+  'Cunt-Boy',
+  'Transgender',
+  'Male-Herm',
+  'None'
+];
+
+export const statusSort: { [key: string]: number } = {
+  crown: 0,
+  looking: 1,
+  online: 2,
+  idle: 3,
+  away: 4,
+  busy: 5,
+  dnd: 6,
+  offline: 7
+};
+
+export const genderSort: { [key: string]: number } = {
+  Female: 0,
+  Male: 1,
+  Herm: 2,
+  Shemale: 3,
+  'Cunt-Boy': 4,
+  Transgender: 5,
+  'Male-Herm': 6,
+  None: 7
+};
+
+function normalizeLabel(s: string | undefined): string {
+  if (!s) return '';
+  return s
+    .toString()
+    .toLowerCase()
+    .replace(/[^a-z0-9]+/g, '');
+}
+
+const canonicalGenderByNormalized: { [key: string]: string } = {};
+Object.keys(genderSort).forEach(k => {
+  canonicalGenderByNormalized[normalizeLabel(k)] = k;
+});
+
+function displayNameForGender(genderValue: number): string {
+  const name = Gender[genderValue as any] as string;
+  if (!name) return 'None';
+  switch (name) {
+    case 'MaleHerm':
+      return 'Male-Herm';
+    case 'Cuntboy':
+      return 'Cunt-Boy';
+    default:
+      return name;
+  }
+}
+
+export function computeGenderPreferenceBuckets(profile: CharacterAnalysis): {
+  match: string[];
+  weakMatch: string[];
+  neutral: string[];
+  weakMismatch: string[];
+  mismatch: string[];
+} {
+  const matches: string[] = [];
+  const weakMatches: string[] = [];
+  const weakMismatches: string[] = [];
+  const mismatches: string[] = [];
+  const neutral: string[] = [];
+
+  const c = profile.character as any;
+
+  Object.values(Gender)
+    .filter(v => typeof v === 'number')
+    .forEach(gv => {
+      let score: Scoring = Scoring.NEUTRAL;
+      const kinkPref = Matcher.getKinkGenderPreference(c, gv as Gender);
+      if (kinkPref === null) {
+        score = Matcher.scoreOrientationByGender(
+          profile.gender,
+          profile.orientation,
+          gv as Gender
```

Comment:
To add to my comment on line 5, there is also an exported [FChatGendermap](https://github.com/Fchat-Horizon/Horizon/blob/983d2e261a3a9efad6dbe54f3041383a7e4a9bd6/learn/matcher-types.ts#L1744) in the matcher code that you can use instead of needing to try and cast values like this.